src/Collections/Sequences/NatSeq.dfy

```diff
@@ -0,0 +1,602 @@
+// RUN: %dafny /compile:0 /noNLarith "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/*******************************************************************************
+*  Original: Copyright (c) 2020 Secure Foundations Lab
+*  SPDX-License-Identifier: MIT
+*  
+*  Modifications and Extensions: Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+*******************************************************************************/
+
+/* The first element of a sequence is the least significant word; the last
+element is the most significant word. */
+
+include "../../NonlinearArithmetic/DivMod.dfy"
+include "../../NonlinearArithmetic/Mul.dfy"
+include "../../NonlinearArithmetic/Power.dfy"
+include "Seq.dfy"
+
+abstract module NatSeq {
```

Comment:
I admit it took me quite a bit of reviewing before I understood what this module was about. :) I think it could use a comment stating this is specifically about interpreting a sequence of numbers positionally with a given base.

I'd also consider giving it a more suggestive name, perhaps `PositionalNat`?

Reply:
Good point, that should definitely be documented more explicitly.  In most languages, this would probably be called `BigNat` or something along those lines.  I'm a bit worried that implies a performant implementation, rather than the specification purpose this library is primarily meant for.